src/main/java/services/ReportImpl.java

```diff
@@ -0,0 +1,42 @@
+package services;
+
+import db.Storage;
+import java.util.List;
+import java.util.Map;
+import services.actions.ActionHandler;
+
+public class ReportImpl implements Report {
+    private final ActionTypeStrategy actionTypeStrategy;
+
+    public ReportImpl(ActionTypeStrategy actionTypeStrategy) {
+        this.actionTypeStrategy = actionTypeStrategy;
+    }
+
+    @Override
+    public String createReport(List<String> records) {
+        StringBuilder report = new StringBuilder();
+        RecordValidation recordValidation = new RecordValidation();
```

Comment:
Let's make this object a class field instead of local one (we don't need to create separate object per each method call